packages/toolpad-app/src/toolpadDataSources/googleSheets/server.ts

```diff
@@ -103,30 +133,34 @@ async function exec(
     version: 'v4',
     auth: client,
   });
-  if (query.spreadsheet) {
-    const response = await sheets.spreadsheets.values.get({
-      spreadsheetId: query.spreadsheet.id,
-      range: query.ranges,
-    });
-    if (response.statusText === 'OK') {
-      const { values } = response.data;
-      if (values && values.length > 0) {
-        const headerRow = values.shift() ?? [];
-        const fields = headerRow.reduce((acc, currValue) => ({ ...acc, [currValue]: '' }), {});
+  if (!query) {
```

Comment:
Not sure how we can enter this conditional, the types seem to suggest this is impossible